### PR TITLE
refactor(ui): replace package:dart_vlc with package:media_kit

### PR DESCRIPTION
**Pull request by alexmercerind**
# Submit a pull request

## CLA

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [x] The code changes follow best practices
- [x] Code changes are tested (add some information if not applicable)

## Description of the pull request

This PR replaces [dart_vlc](https://pub.dev/packages/dart_vlc) with [media_kit](https://pub.dev/packages/media_kit) (used on Microsoft Windows & GNU/Linux in [stream_chat_flutter](https://pub.dev/packages/stream_chat_flutter)). media_kit has major benefits over dart_vlc e.g. better stability, improved performance, less bundle size etc.

To simplify & reduce the total size of implementation, [video_player_media_kit](https://pub.dev/packages/video_player_media_kit) is used. It automatically adds support for unsupported platforms to video_player. This means existing video_player-based implementation is automatically shared between all platforms without any additional implementation, wrapper or boilerplate.

[video_player_media_kit is part of media_kit project itself](https://github.com/media-kit/media-kit/tree/main/video_player_media_kit), where it receives similar support & maintenance.

It also happens to be sponsored by GetStream.IO!

References:
- https://pub.dev/packages/media_kit
- https://pub.dev/packages/video_player_media_kit
- https://github.com/media-kit/media-kit

### CODE REVIEW

**Comment by alexmercerind** [comment]
I tested the [stream_chat_flutter/example app](https://github.com/GetStream/stream-chat-flutter/tree/develop/packages/stream_chat_flutter/example) on Windows & GNU/Linux after changes, videos seem to play just fine.

```yaml
name: stream_chat_flutter_example
description: A new Flutter project.
publish_to: none
version: 1.0.0+1

environment:
  sdk: ">=3.0.0 <4.0.0"
  flutter: ">=3.10.0"

dependencies:
  collection: ^1.15.0
  cupertino_icons: ^1.0.4
  flutter:
    sdk: flutter
  responsive_builder: ^0.7.0
  stream_chat_flutter:
    path: ../../stream_chat_flutter
  stream_chat_localizations:
    path: ../../stream_chat_localizations
  stream_chat_persistence:
    path: ../../stream_chat_persistence

dependency_overrides:
  stream_chat_flutter:
    path: ../../stream_chat_flutter
  stream_chat_localizations:
    path: ../../stream_chat_localizations
  stream_chat_persistence:
    path: ../../stream_chat_persistence

flutter:
  uses-material-design: true
  assets:
    - assets/background_doodle.png
```

File: packages/stream_chat_flutter/lib/src/stream_chat.dart [hunk 1]
```diff
@@ -95,52 +95,73 @@ class StreamChatState extends State<StreamChat> {
...
   void initState() {
     super.initState();
-    // Ensures that VLC only initializes in real desktop environments
-    if (!isTestEnvironment && isDesktopVideoPlayerSupported) {
-      VlcManager.instance.initialize();
+    // package:video_player does not support Microsoft Windows & GNU/Linux.
+    // package:video_player_media_kit bridges support for these platforms using
+    // package:media_kit.
+    if (!_mediaKitInitialized && isDesktopVideoPlayerSupported) {
+      _mediaKitInitialized = true;
```
Comment (esarbanis) [inline packages/stream_chat_flutter/lib/src/stream_chat.dart:108]: Shouldn't this happer after `ensureInitialized` is invoked, as mentioned in the variable's comment?

**Review by esarbanis — CHANGES_REQUESTED** [review]
I really think we should first deprecate `FullScreenMedia*` classes before removing them. They are part of our public interface, and if we remove them like that. will probably break users' apps.

File: packages/stream_chat_flutter/lib/src/fullscreen_media/full_screen_media_builder.dart [hunk 2]
```diff
@@ -1,30 +1,10 @@
...
-/// This widget makes use of dart's conditional imports to ensure that Stream's
-/// desktop implementation of StreamFullScreenMedia is not imported when
-/// building applications that target web. Additionally, this widget ensures
-/// that applications targeting mobile platforms do not build the version of
-/// StreamFullScreenMedia that targets desktop platforms (even though
-/// `package:dart_vlc` technically supports iOS).
-/// {@endtemplate}
+/// A full screen media widget
+@Deprecated('Use StreamFullScreenMedia instead.')
 class StreamFullScreenMediaBuilder extends StatelessWidget {
```
Comment (alexmercerind) [inline packages/stream_chat_flutter/lib/src/fullscreen_media/full_screen_media_builder.dart:6]: I preserved & deprecated `StreamFullScreenMediaBuilder`.

`StreamFullScreenMedia` is recommended replacement & directly returned by this widget.

File: packages/stream_chat_flutter/lib/src/stream_chat.dart (hunk 1 above)
Comment (alexmercerind) [reply packages/stream_chat_flutter/lib/src/stream_chat.dart]: I made the changes upstream to internally check for multiple initializations: https://github.com/media-kit/media-kit/commit/6516ef12cb7c117ad670c64901efdfa6c344af44.

This eliminates the requirement of previous additional check & static variable.

**Comment by alexmercerind** [comment]
> I really think we should first deprecate FullScreenMedia* classes before removing them. They are part of our public interface, and if we remove them like that. will probably break users' apps.

I hope the changes are according to your expectation. 

Thanks!

**Comment by jnelle** [comment]
@alexmercerind any updates on this?

**Pull request closed without merging**